client/network/sync/src/lib.rs

```diff
@@ -1421,22 +1472,56 @@ where
 			.map_err(|error: codec::Error| error.to_string())
 	}
 
-	fn poll(&mut self, cx: &mut std::task::Context) -> Poll<PollResult<B>> {
+	fn poll(
+		&mut self,
+		cx: &mut std::task::Context,
+	) -> Poll<PollBlockAnnounceValidation<B::Header>> {
 		while let Poll::Ready(Some(event)) = self.service_rx.poll_next_unpin(cx) {
 			match event {
 				ToServiceCommand::SetSyncForkRequest(peers, hash, number) => {
 					self.set_sync_fork_request(peers, &hash, number);
 				},
+				ToServiceCommand::RequestJustification(hash, number) =>
+					self.request_justification(&hash, number),
+				ToServiceCommand::ClearJustificationRequests => self.clear_justification_requests(),
+				ToServiceCommand::BlocksProcessed(imported, count, results) => {
+					for result in self.on_blocks_processed(imported, count, results) {
```

Comment:
`on_blocks_processed` doesn't really need to stay in the trait of `Sync` anymore? Aka it could only be a function of `ChainSync` or?